### ISSUE TITLE
Frequent sound crackle and audio distortion - Linux

### ISSUE THREAD

**Issue by 7oxicshadow** [loganmc10/GLupeN64#105]
The best examples of this are conkers bad fur day and diddy kong racing. If you watch the intros on both games the crackle should be heard. Diddy Kong Racing is a good example of audio distortion, when watching the intro and timber is on the hovercraft the music audio doesnt sound correct.

I have tried other cores and they all seem to be fine so it suggests its unique to this one.

Its almost as if the audio buffer is under flowing.

**Comment by 7oxicshadow** [comment]
Quick update. Setting VSYNC to off results in normal audio with no crackle or distortion

**Comment by loganmc10** [comment]
Yep I experience this on my Linux laptop as well, quite annoying. I doesn't happen on Android for me, and I don't think it happens in Windows 

I've played around with it a bit in the past but I've never been able to solve it. It's quite noticeable in the intro to goldeneye for me.

**Comment by loganmc10** [comment]
Does it happen with mupen64plus-libretro for you?

**Comment by 7oxicshadow** [comment]
mupen64plus-libretro works fine in terms of audio. No distortion with VSYNC on or off.

As for GLupen64 it almost as if the audio process is not being run during wait period for VSYNC.

**Comment by loganmc10** [comment]
Did you turn vsync back on? For me, the intro to GoldenEye sounds the exact same in GLupeN64 and mupen64plus-libretro (crackly)

**Comment by 7oxicshadow** [comment]
Just checked again. 100% sure that vsync is on with no audio problems on mupen64plus-libretro (1.3.6 GIT a292) on goldeneye (or any other game that i have noticed).

Just incase:

i7 6700k @ 4.5ghz
16gig ram
GTX 770 (370.28 propitiatory driver)

**Comment by 7oxicshadow** [comment]
Interesting. If I use alsa-thread as the audio driver the problem goes away on GLupen64

**Comment by loganmc10** [comment]
Ya maybe it's related to ALSA audio, Android uses opensl and Windows uses whatever Windows uses, so that would explain why it only happens on Linux. 

@orbea have you ever experienced this on Linux?

**Comment by loganmc10** [comment]
I tried the Pulse Audio driver and it also fixes this issue for me. I experience the same thing in mupen64plus-libretro. It is probably dependant on how fast the system is and how well it can keep everything in sync. I have a fairly old system so it is pretty noticeable.

I'm not really an audio guy, so I don't think I'd be able to fix this, but I did add a page to the Wiki about this problem:
https://github.com/loganmc10/GLupeN64/wiki/Audio-Tips

That is probably going to need to be sufficient unless someone else thinks they can solve this problem
